src/java/org/streamingpool/core/support/StreamSupport.java

```diff
@@ -41,8 +44,14 @@
 
     <T> OngoingLazyProviding<T> provide(StreamCreator<T> reactStream);
 
+    default Set<StreamId<?>> getSubgraphStartingFrom(StreamId<?> sourceId) {
+        return dependencyService().dependencyTree().getAncestorsFrom(sourceId);
+    }
+
     ProvidingService providingService();
 
+    InstrumentationService dependencyService();
```

Comment:
I wonder why exposing the dependency service here and then having a default method. ... Why not hiding the dependecy service and just implementing the 'subgraph...' method in the subclass...?

Reply:
just flexibility. So the tests can access all the services directly if they need to (like providingService). Also because we could have default methods that will allow composition of interfaces, like RxSupport, etc.. and the final one should only implement the getter of the services :)